rskj-core/src/main/java/co/rsk/config/RemascConfigFactory.java

```diff
@@ -34,19 +38,39 @@ public class RemascConfigFactory {
     private static final Logger logger = LoggerFactory.getLogger("RemascConfigFactory");
 
     private ObjectMapper mapper;
-    private String configPath;
+    private final JsonNode configNode;
 
     public RemascConfigFactory(String remascConfigFile) {
         this.mapper = new ObjectMapper();
-        this.configPath = remascConfigFile;
+        logger.info("RemascConfigFactory remascConfigFile = {} ", remascConfigFile);
+
+        try (InputStream is = RemascConfigFactory.class.getClassLoader().getResourceAsStream(remascConfigFile)) {
+            String sis = new BufferedReader(new InputStreamReader(is, StandardCharsets.UTF_8))
+                    .lines()
+                    .collect(Collectors.joining("\n"));
+
+            logger.info("RemascConfigFactory sis = {} ", sis);
+
+            configNode = JacksonParserUtil.readTree(mapper, sis);
+
+            logger.info("RemascConfigFactory node = {} ", configNode);
+        } catch (Exception ex) {
+            logger.error("Error reading REMASC", ex);
+            throw new RemascException("Error reading REMASC [" + remascConfigFile + "]: ", ex);
+        }
     }
 
     public RemascConfig createRemascConfig(String config) {
         RemascConfig remascConfig;
 
-        try (InputStream is = RemascConfigFactory.class.getClassLoader().getResourceAsStream(this.configPath)) {
-            JsonNode node = JacksonParserUtil.readTree(mapper, is);
-            remascConfig = JacksonParserUtil.treeToValue(mapper, node.get(config), RemascConfig.class);
+        logger.info("createRemascConfig config = {} ", config);
+
+        try {
+            logger.info("createRemascConfig node.get(config) = {} ", configNode.get(config));
+
+            remascConfig = JacksonParserUtil.treeToValue(mapper, configNode.get(config), RemascConfig.class);
```

Comment:
It seems that `RemascConfig` is immutable and permanent. What about parsing this only once and storing results, for instance, in `ConcurrentHashMap<String, RemascConfig>` for later usage?

Reply:
Addressed